core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -140,6 +147,64 @@ export class AdminBackendApiService {
       });
     });
   }
+
+  private _postAdminActionAsync(action:string, payload:Object): Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, { action, ...payload }).toPromise()
+      .then(response => {
+        return response;
+      }, errorResonse => {
+        return errorResonse.error.error;
+      });
+  }
+
+  async startNewJobAsync(jobType: string): Promise<void> {
+    let action = 'start_new_job';
+    let payload = {
+      job_type: jobType
+    };
+    return this._postAdminActionAsync(action, payload);
+  }
+
+  async cancelJobAsync(jobId: string, jobType: string): Promise<void> {
+    let action = 'cancel_job';
+    let payload = {
+      job_id: jobId,
+      job_type: jobType
+    };
+    return this._postAdminActionAsync(action, payload);
+  }
+
+  async startComputationAsync(computationType: string): Promise<void> {
+    let action = 'start_computation';
+    let payload = {
+      computation_type: computationType
+    };
+    return this._postAdminActionAsync(action, payload);
+  }
+
+  async stopComputationAsync(computationType: string): Promise<void> {
+    let action = 'stop_computation';
+    let payload = {
+      computation_type: computationType
+    };
+    return this._postAdminActionAsync(action, payload);
+  }
+
+  async fetchJobOutputAsync(jobId: string): Promise<string[]> {
+    let adminJobOutputUrl = this.urlInterpolationService.interpolateUrl(
+      AdminPageConstants.ADMIN_JOB_OUTPUT_URL_TEMPLATE, {
+        jobId: jobId
+      });
+    return new Promise((resolve, reject) => {
+      this.http.get<JobOutput>(
+        adminJobOutputUrl).toPromise().then(response => {
+        resolve(response.output.sort() || []);
```

Comment:
We still need to fix this. (Assume output isn't a list? then it will fail here, right?

`resolve(Array.isArray(response.output) ? response.output.sort() : [])`; Also, add a test case for all the items in the list are sorted.